packages/core/core-flows/src/cart/workflows/complete-cart.ts

```diff
@@ -116,9 +123,26 @@ export const completeCartWorkflow = createWorkflow(
     })
 
     // If order ID does not exist, we are completing the cart for the first time
-    const order = when("create-order", { orderId }, ({ orderId }) => {
-      return !orderId
-    }).then(() => {
+    const order = when(
+      "create-order",
+      { orderId, paymentId: input.payment_id },
+      ({ orderId, paymentId }) => {
```

Comment:
```suggestion
      { orderId },
      ({ orderId }) => {
```

This input is only for the `when` block. No need to add it here